### DISCUSSION TITLE
Middleware in KaTeX?

### DISCUSSION THREAD

**Discussion by ahmedelq** [KaTeX/KaTeX#3619] [Q&A] [answered]
Hi, I was wondering if there's a way to write my custom middleware.

I basically want to write some kind of an extension to add my modify the symbols or to add my own, modify the style and add a custom font. 

I also want to share this middleware/extension/plugin with other people. 
I know that KaTeX ships with some extensions, but how can I write my own?

Does KaTeX support something like this?
Thanks

**Comment by edemaine** [comment, marked as answer]
Take a look at https://github.com/KaTeX/KaTeX/tree/main/contrib/mhchem for an example of defining custom macros. There's no support for adding actual symbols/fonts though.
